### PR TITLE
Rachel/user creation

### PR DESCRIPTION
**Pull request by blakerach1**
## Describe your changes

- Added 'rest_framework' to INSTALLED_APPS in settings.py - this is required after installing Django REST Framework.
- User app created, app to settings.py

- CustomUser(AbstractUser) model added to users>models.py, model also added to settings.py
> I have also added a custom validator to ensure case-insensitive uniqueness of usernames. 
> I've added the required acceptance of T's & C's in order for User to create an account. 
> Added permission on CustomUserList view - I want only the Users list to be visible by admin
> Added permission on CustomUserDetail view - Only user details will be returned to admin or the user themselves. A user cannot access the details of another user. 

- Created CustomUserSerializer - passwords hashed and set up as write only
- Created view to list all users, create new users and view details of a single user

- Added urls for Users, and registered with the global project DiversiTech>urls.py

**Tests in insomnia:**
GET /users/ - returned 200 OK
GET /user/1/ - returned 200 OK
POST /users/ - returned 200 OK

- Login / Logout established using built in url path 'api-auth/'
- I have set up token authentication

**Tests in insomnia:**
POST /api-token-auth/ - returned 200 ok & token for superuser

- I have one migration file (separately commited to this branch last). 

- added CustomUser model to in-built admin panel
- small formatting fix in under CustomUser model
- added custom obtain_auth_token view which now returns token, id and email when user provides valid username and password

- Validators are not enabled by default when using Django Rest Framework, so I have enabled the inbuild password validators on the user serializer. Min length, can't be similar to username, firstname or lastname, not a common password (like password123) and not all numeric.  

- I've also amended the serializer to refer to the User model via global settings, rather than referencing it via local path. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## What kind of feedback do you need?

Thorough assessment?

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have recommended where to start and how to proceed with the review

### CODE REVIEW

**Review by shayzimm — APPROVED** [review]
This looks great to me, Rachel - I have a few questions about the user table and the difference between a superuser and staff? I'll send in Slack :)

File: DiversiTech/users/validators.py
Comment (Lyndallh) [inline DiversiTech/users/validators.py:1]: This is cool. I'm looking forward to seeing it in action.

File: DiversiTech/users/views.py
Comment (Lyndallh) [inline DiversiTech/users/views.py:1]: Slack message sent re: returning additional user details with token.

**Review by Lyndallh — COMMENTED** [review]
comments left on individual files

**Review by ktechgau — APPROVED** [review]

**Review by awesomeann — APPROVED** [review]

**Pull request merged**